code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm

```diff
@@ -0,0 +1,111 @@
+///To be used when there is the need of an atmos connection without repathing everything (eg: cryo.dm)
+/obj/machinery/atmospherics/components/unary/machine_connector
+	name = "machine connector"
+	desc = "Internal connector to interface with the pipes."
+
+	can_unwrench = FALSE
+
+	use_power = NO_POWER_USE
+	layer = GAS_FILTER_LAYER
+	hide = TRUE
+	shift_underlay_only = FALSE
+
+	pipe_flags = PIPING_ONE_PER_TURF
+	resistance_flags = INDESTRUCTIBLE
+
+	///Reference to the machine we are connected to
+	var/obj/machinery/connected_machine
+
+/obj/machinery/atmospherics/components/unary/machine_connector/Initialize(mapload, process = TRUE, setdir, init_dir = ALL_CARDINALS, obj/machinery/connecting_machine, direction)
```

Comment:
you should sort these

Reply:
i'll sort the last two only because the others are from atmosmachinery New(), i had to add them due to an issue with the create and destroy test

Reply:
we have this stuff in new? guh

Reply:
shitcode strikes again (there's a reason for it I'm pretty sure it's just dumb and shouldn't exist)

Reply:
Also this is cursing me badly because I'm having issues with the connector not working properly so I'll have to remake some part of this